spec.emu

```diff
@@ -0,0 +1,1319 @@
+    <li>The <dfn id="json-names"><code>names</code> field</dfn> is an optional list of symbol names which may be used by the mappings field.</li>
+    <li>The <dfn id="json-mappings"><code>mappings</code> field</dfn> is a string with the encoded mapping data (see section <emu-xref href="#sec-mappings"></emu-xref>).</li>
+    <li>The <dfn id="json-ignoreList"><code>ignoreList</code> field</dfn> is an optional list of indices of files that should be considered third party code, such as framework code or bundler-generated code. This allows developer tools to avoid code that developers likely don't want to see or step through, without requiring developers to configure this beforehand. It refers to the sources field and lists the indices of all the known third-party sources in the source map. Some browsers may also use the deprecated `x_google_ignoreList` field if `ignoreList` is not present.</li>
+  </ul>
+
+  <emu-clause id="sec-decoding-source-maps">
+    <h1>Decoding source maps</h1>
+
+    <p>A <dfn id="decoded-source-map-record" variants="Decoded Source Map Records">Decoded Source Map Record</dfn> has the following fields:</p>
+    <emu-table id="table-decoded-source-map-fields" caption="Fields of Decoded Source Map Records">
+      <table>
+        <thead>
+          <tr>
+            <th>
+              Field Name
+            </th>
+            <th>
+              Value Type
+            </th>
+          </tr>
+        </thead>
+        <tr>
+          <td>[[File]]</td>
+          <td>a String or *null*</td>
+        </tr>
+        <tr>
+          <td>[[Sources]]</td>
+          <td>a List of Decoded Source Records</td>
+        </tr>
+        <tr>
+          <td>[[Mappings]]</td>
+          <td>a List of Decoded Mapping Records</td>
+        </tr>
+      </table>
+    </emu-table>
+
+    <p>A <dfn id="decoded-source-record" variants="Decoded Source Records">Decoded Source Record</dfn> has the following fields:</p>
+    <emu-table id="table-decoded-source-fields" caption="Fields of Decoded Source Records">
+      <table>
+        <thead>
+          <tr>
+            <th>
+              Field Name
+            </th>
+            <th>
+              Value Type
+            </th>
+          </tr>
+        </thead>
+        <tr>
+          <td>[[URL]]</td>
+          <td>a URL or *null*</td>
+        </tr>
+        <tr>
+          <td>[[Content]]</td>
+          <td>a String or *null*</td>
+        </tr>
+        <tr>
+          <td>[[Ignored]]</td>
+          <td>a Boolean</td>
+        </tr>
+      </table>
+    </emu-table>
+
+    <emu-clause id="sec-ParseSourceMap" type="abstract operation">
+      <h1>
+        ParseSourceMap (
+          _string_: a String,
+          _baseURL_: an URL,
+        ): a Decoded Source Map Record
+      </h1>
+      <dl class="header"></dl>
+      <emu-alg>
+        1. Let _json_ be Call(%JSON.parse%, *null*, « _string_ »).
```

Comment:
Technically this would require implementors to follow exact JSON.parse semantics. And below to follow ECMAScript specific semantics (`HasOwnProperty`).

Should we instead use a "host-defined" or "implementation-defined" operation that returns a JSON `object`?

IMO the "base data types" the source map spec should operate on are the ones defined by JSON: `object`, `array`, `string`, `number`, `true`, `false` and `null`. And we assemble spec records from these.

Reply:
Note that this was already the case, as we were using [parse JSON bytes to an Infra value](https://infra.spec.whatwg.org/#parse-a-json-string-to-an-infra-value) which internally does `Call(%JSON.parse%, *null*, « _string_ »)`.

What about adding a sentence in the new "Notational Conventions" section saying that, when our spec references ECMA-262 operations, those are to be interpreted as running in a fresh not-user-modified environment? Then whether you actually use JavaScript object or another representation is not observable (even though I think it would not be observable even if the user messes up with `Object.prototype`, because nether `JSON.parse` nor `HasOwnProperty` on the object it returns are affected by it)

Reply:
The infra spec uses `%JSON.parse%` but then also immediately converts the resulting JS object back into a "infra value" (unordered map) so "callers" don't have to use ECMA-262 operations.

I'm just wondering if the spec will be more readable if we define a small `Get` and `Has` on JSON values ourselves, rather than use `HasOwnProperty` and ECMA-262 `Get` on real JS objects.

No strong opinion though, so if you and Asumu prefer to use ECMA-262 operations, I'm also fine with that.